src/mscorlib/src/System/String.cs

```diff
@@ -121,30 +113,23 @@ public sealed class String : IComparable, ICloneable, IConvertible, IEnumerable
             Contract.Ensures(Contract.Result<String>() != null);
             Contract.EndContractBlock();
 
-            if (separator == null)
-                separator = String.Empty;
-
             using(IEnumerator<T> en = values.GetEnumerator()) {
                 if (!en.MoveNext())
                     return String.Empty;
 
                 StringBuilder result = StringBuilderCache.Acquire();
-                if (en.Current != null) {
-                    // handle the case that the enumeration has null entries
-                    // and the case where their ToString() override is broken
-                    string value = en.Current.ToString();
-                    if (value != null)
-                        result.Append(value);
+                T currentValue = en.Current;
+
+                if (currentValue != null) {
+                    result.Append(currentValue.ToString());
                 }
 
                 while (en.MoveNext()) {
+                    currentValue = en.Current;
```

Comment:
Never mind. When I was looking over this before, I was thinking about a bad behaving implementation of `Current` that happens to throw when called here (already pretty unlikely). Previously, the separator would be written to the `StringBuilder` before it threw, but after this change it would throw before the separator is appended. But since the `StringBuilder` isn't returned until the end of the method it shouldn't matter.